x-pack/plugins/apm/server/routes/correlations/queries/fetch_duration_ranges.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  SPAN_DURATION,
+  TRANSACTION_DURATION,
+} from '../../../../common/elasticsearch_fieldnames';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import { Setup } from '../../../lib/helpers/setup_request';
+import { getCommonCorrelationsQuery } from './get_common_correlations_query';
+import { Environment } from '../../../../common/environment_rt';
+
+export const fetchDurationRanges = async ({
+  rangeSteps,
+  setup,
+  start,
+  end,
+  environment,
+  kuery,
+  query,
+  eventType,
+}: {
+  rangeSteps: number[];
+  setup: Setup;
+  start: number;
+  end: number;
+  environment: Environment;
+  kuery: string;
+  query: estypes.QueryDslQueryContainer;
+  eventType: ProcessorEvent;
+}): Promise<Array<{ key: number; doc_count: number }>> => {
+  const { apmEventClient } = setup;
+
+  const ranges = rangeSteps.reduce(
+    (p, to) => {
+      const from = p[p.length - 1].to;
+      p.push({ from, to });
+      return p;
+    },
+    [{ to: 0 }] as Array<{ from?: number; to?: number }>
+  );
```

Comment:
Can't you do like this?
```suggestion
  const ranges = rangeSteps.reduce<Array<{ from?: number; to?: number }>
>(
    (p, to) => {
      const from = p[p.length - 1].to;
      p.push({ from, to });
      return p;
    },
    [{ to: 0 }] );
```

Reply:
I don't think either one is much better than the other, so I will just leave it as is.